src/platform/site-wide/user-nav/components/PersonalizationDropdown.jsx

```diff
@@ -15,6 +17,8 @@ const recordProfileEvent = recordNavUserEvent('profile');
 const recordAccountEvent = recordNavUserEvent('account');
 
 export class PersonalizationDropdown extends React.Component {
+  profileUrl = () => (this.props.useProfile2 ? '/profile-2' : '/profile');
```

Comment:
Needed to set this as a function rather than a simple property. When it was a property it would always be set to `'/profile'` because `props.useProfile2` starts as `false` and then updates after initial render.